src/coredump/coredump.c

```diff
@@ -1462,6 +1478,33 @@ static int process_kernel(int argc, char* argv[]) {
                 log_open();
         }
 
+        r = in_same_namespace(getpid_cached(), context.pid, NAMESPACE_PID);
+        if (r < 0)
+                log_debug_errno(r, "Failed to check pidns of crashing process, ignoring: %m");
+
+        if (r == 0 && getenv_bool("SYSTEMD_COREDUMP_ALLOW_NAMESPACE_CHANGE") > 0) {
```

Comment:
I'm trying to ~~break~~ write some tests for this and I'm slightly confused _where_ this env variable is supposed to be set? It can't be set in the `systemd-coredump@.service` as that's too late (that service is spawned once the `kernel.core_pattern` sd-coredump instance is done processing the coredump), it won't propagate to the `core_pattern` binary environment from the manager, and setting it in the crashed process' environment doesn't work either, since the `core_pattern` binary is executed outside of that environment. So in all cases this `getenv()` will return `-ENXIO`.

Reply:
Well, for I testing I pointed to core_pattern to my own script which sets env variable and just executes systemd-coredump (this is roughly what actually what customer is going to do as well).

Reply:
Here is what I used,

```
$ sysctl kernel.core_pattern
kernel.core_pattern = |/usr/local/bin/coredumper %P %u %g %s %t %c %h

$ cat /usr/local/bin/coredumper
#!/bin/bash

export SYSTEMD_COREDUMP_ALLOW_NAMESPACE_CHANGE=1
exec /usr/lib/systemd/systemd-coredump "$@"
```